recurring_payments_stripe/models/account_move.py

```diff
@@ -0,0 +1,169 @@
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # def stripe_pay_invoice(self):
+    #     """Paga la factura en Stripe si `is_recurrent` está activado en la suscripción."""
+    #     for invoice in self:
+    #         subscription = invoice.invoice_date and self.env[
+    #             "sale.subscription"
+    #         ].search([("code", "=", invoice.invoice_date)], limit=1)
+    #         if (
+    #             subscription
+    #             and subscription.is_recurrent
+    #             and subscription.stripe_customer
+    #         ):
+    #             provider = self.env["payment.provider"].search(
+    #                 [("code", "=", "stripe")],
+    #             )
+    #             stripe.api_key = provider.stripe_secret_key
+    #             token = self.env["payment.token"].search(
+    #                 [("provider_id", "=", provider.id)]
+    #             )
+    #             try:
+    #                 # Crea un PaymentIntent en Stripe para el monto de la factura
+    #                 payment_intent = stripe.PaymentIntent.create(
+    #                     # Stripe maneja montos en centavos
+    #                     amount=int(invoice.amount_total * 100),
+    #                     currency=invoice.currency_id.name.lower(),
+    #                     customer=token.provider_ref,
+    #                     payment_method=token.stripe_payment_method,
+    #                     payment_method_types=["card"],
+    #                     # Para pagos automáticos sin intervención del usuario
+    #                     off_session=True,
+    #                     # Confirmar el PaymentIntent inmediatamente
+    #                     confirm=True,
+    #                     metadata={"odoo_invoice_id": invoice.id},
+    #                 )
+
+    #                 # Confirmar el pago y actualizar el estado de la factura en Odoo
+    #                 if payment_intent["status"] == "succeeded":
+    #                     invoice.action_post()
+    #                     invoice.payment_state = "paid"
+    #                 else:
+    #                     raise Exception(
+    #                         f"Error en el pago de Stripe: {payment_intent['status']}"
+    #                     )
+
+    #             except stripe.StripeError as e:
+    #                 raise UserError(f"Stripe error: {e.user_message or str(e)}")
+
+    # def action_post(self):
+    #     """Sobreescribe el método `action_post` para procesar el pago automático si `is_recurrent` está activo."""
+    #     res = super(AccountMove, self).action_post()
+    #     for invoice in self:
+    #         subscription = self.env["sale.subscription"].search(
+    #             [("code", "=", invoice.invoice_date)], limit=1
+    #         )
+    #         if subscription and subscription.is_recurrent:
+    #             invoice.stripe_pay_invoice()
+    #     return res
+
+    def action_register_payment(self):
```

Comment:
In this case i think it would be better to use context to separate if the payment is been processed by the cron or manually.